### PR TITLE
[GH-1334] Move `source` Interface and Implementation into new namespace

### PR DESCRIPTION
**Pull request by ehigham**
RR: https://broadinstitute.atlassian.net/browse/GH-1334
Move `source` interface, helper functions and implementations into new `wfl.source` namespace
Move some source-specific integration tests to a new `source-test` namespace
Remove IO from datarepo namespace

### CODE REVIEW

**Review by rhiananthony — APPROVED** [review]

**Review by tbl3rd — APPROVED** [review]
I didn't check that all the moved code was copied correctly.

**Review by tbl3rd — COMMENTED** [review]
Also missed the last couple of commits.

File: makerules/modules.mk [hunk 1]
```diff
@@ -38,11 +37,10 @@ all: $(MAKE_TARGETS)
...
 $(PREBUILD):    $(shell eval $(MKDIR) $(DERIVED_MODULE_DIR))
-$(FORMAT):      $(PREBUILD)
-$(BUILD):       $(FORMAT)
+$(LINT):        $(PREBUILD)
+$(BUILD):       $(PREBUILD)
 $(UNIT):        $(BUILD)
 $(INTEGRATION): $(BUILD)
-$(LINT):        $(BUILD)
 $(IMAGES):      $(BUILD)
 $(SYSTEM):      $(BUILD)
```
Comment (tbl3rd) [inline makerules/modules.mk:45]: Retain the `format` target for easy reformatting?
I'm OK with removing the dependencies though.

File: api/src/wfl/tsv.clj [hunk 2]
```diff
@@ -77,7 +77,7 @@
   [table]
   (let [writer (StringWriter.)]
     (write table writer)
-    (.toString writer)))
+    (str writer)))
```
Comment (tbl3rd) [inline api/src/wfl/tsv.clj:80]: Kinda prefer `.toString` here as well, but OK if that works.
(I know some linter complains.)

File: makerules/modules.mk (hunk 1 above)
Comment (ehigham) [reply makerules/modules.mk:45]: That's what the clojure alias is for - make isn't for general scripting

Comment (tbl3rd) [reply makerules/modules.mk:45]: Then maybe restore this comment?

# Run `clojure -M:format` in this directory when this fails.

Even better: have a format check failure describe the remedy.

Comment (tbl3rd) [reply makerules/modules.mk:45]: Love markdown.

**Pull request merged**